recipes/dirent/all/conanfile.py

```diff
@@ -0,0 +1,32 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class DirEntConan(ConanFile):
+    name = "dirent"
+    description = "Dirent is a C/C++ programming interface that allows programmers to retrieve information about " \
+                  "files and directories under Linux/UNIX"
+    topics = ("conan", "dirent", "directory", "file system")
+    homepage = "https://github.com/tronkko/dirent"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    settings = "compiler"
+    no_copy_source = True
+
+    _source_subfolder = "source_subfolder"
+
+    def configure(self):
+        if self.settings.compiler != "Visual Studio":
```

Comment:
not quite sure. We can use clang for instance. I would the best check here is os == "Windows"

Reply:
Doen't clang have a dirent implementation? mingw has.
How about `self.settings.os != "Windows" or self.settings.compiler != "gcc"`

Reply:
```
λ clang-cl.exe test_package.cpp -o test
test_package.cpp(31,10): fatal error: 'dirent.h' file not found
#include <dirent.h>
         ^~~~~~~~~~
1 error generated.
```

nope, only mingw has an implementation.